ember.s

```diff
@@ -147,9 +147,42 @@ not_poke:
 
 	call verify_end
 	call di
-	jmp short shell
+	jmp shell
 
 not_run:
+	cmp al, 'w'
+	jnz short .not_write
+	mov cl, 0x00
+	jmp short .continue
+.not_write:
+	cmp al, 'r'
+	mov cl, 0x01
+	jnz short not_rw
+.continue:	
+        mov bx, di ; move the read location
+	call readhexword
+        jc short parse_error
+	shl eax, 16
+	call readhexword
+        jc short parse_error
+	; eax should hold our lba now
+
+	mov di, bx
+	push es
+	push cs
+	pop es
+	cmp cl, 0x01
+	je .read
+	int i_diskwrite
+	jmp .continue2
+.read:
+	int i_diskread
+.continue2:
```

Comment:
Have you considered using a single `int` instruction whose interrupt number gets modified where you set `cl` now?